pkg/cloudprovider/aws/instancetype.go

```diff
@@ -48,7 +50,16 @@ func (i *InstanceType) Offerings() []cloudprovider.Offering {
 }
 
 func (i *InstanceType) OperatingSystems() sets.String {
-	return sets.NewString("linux")
+	// https://docs.aws.amazon.com/eks/latest/userguide/windows-support.html#windows-support-prerequisites
+	if functional.HasAnyPrefix(i.Name(), "c3", "c4", "d2", "i2", "m6a", "r3") {
```

Comment:
I think there may need to be some additional filtering done here...
1 ) mac* should probably cause errors in general if this is not already catered for
2) any of the graviton series instances (arm) should probably be hard-coded to linux in the same way until ARM support is around (https://eksctl.io/usage/arm-support/) unless there's something i'm unaware of, so this would be t4g, m6g, a1, c7g, c6g, c6gn, r6g, x2gd, im4gn, is4gen and others if i missed them